### PR TITLE
WT-12083 Extend conn_api.c to support new get_configuration method

### PR DESCRIPTION
**Pull request by y123456yz**
get connection configure string surport, connection api surport get_configure.

### CODE REVIEW

**Comment by y123456yz** [comment]
The background is as follows:
>  db.adminCommand( { setParameter : 1, "wiredTigerEngineRuntimeConfig" : "io_capacity=(total=110M)"})
{ "was" : "", "ok" : 1 }
> 
> db.adminCommand( { setParameter : 1, "wiredTigerEngineRuntimeConfig" : "eviction=(threads_min=8, threads_max=12)"})
{ "was" : "io_capacity=(total=110M)", "ok" : 1 }
> 
> db.adminCommand( { setParameter : 1, "wiredTigerEngineRuntimeConfig" : "verbose=[api:5]"})
{ "was" : "eviction=(threads_min=8, threads_max=12)", "ok" : 1 }
> 
> db.adminCommand( { getParameter : "1", wiredTigerEngineRuntimeConfig : 1  } )
{ "wiredTigerEngineRuntimeConfig" : "verbose=[api:5]", "ok" : 1 }
> 

image

When we modify the configuration information of WT multiple times, Mongo server can only retain the last result and cannot obtain all modification records.

**Comment by y123456yz** [comment]
after wt merge this PR's code, I will add a PR to mongo server, perfect "db.adminCommand( { getParameter : "1", wiredTigerEngineRuntimeConfig : 1 } )", By gett_configure api to obtain complete wt configuration information.

**Comment by y123456yz** [comment]
WT-12083

File: src/conn/conn_api.c
Comment (clarissecheah) [inline src/conn/conn_api.c:1328]: Unnecessary deletion here

File: src/conn/conn_api.c
Comment (clarissecheah) [inline src/conn/conn_api.c:1336]: Let's rename to `get_configuration`

**Comment by y123456yz** [comment]
@clarissecheah 
I have change the code, please review again.
thanks.

File: examples/c/ex_all.c
Comment (clarissecheah) [inline examples/c/ex_all.c:1048]: Looks like we missed changing two instances of configure -> configuration here and in `wiredtiger.in:2493`

Comment (y123456yz) [reply examples/c/ex_all.c]: hi, @clarissecheah 
please review again.
thanks.

File: src/conn/conn_api.c
Comment (clarissecheah) [inline src/conn/conn_api.c:1336]: Didn't pick this one up earlier - sorry! This should be: 

`__conn_get_configuration` instead of `__conn_get_home`

I believe this will fix the `s_all` issue we have in the evergreen patch failure.

**Comment by clarissecheah** [comment]
Hi @y123456yz, one more thing - there's currently a known issue being fixed in clang analyzer that's causing the Evergreen [patch failure we see.](https://parsley.mongodb.com/evergreen/wiredtiger_ubuntu2004_asan_clang_analyzer_patch_3a602b8a22cfedaa2f796fe3580bd0880e2fcee8_657c059fa4cf475058a98567_23_12_18_02_11_31/0/task?bookmarks=0,276) Could you merge the develop branch into your branch when you fix the comment I mentioned above? I'll re-review after, but this PR looks good to go after that. Thanks!

**Comment by y123456yz** [comment]
> Hi @y123456yz, one more thing - there's currently a known issue being fixed in clang analyzer that's causing the Evergreen [patch failure we see.](https://parsley.mongodb.com/evergreen/wiredtiger_ubuntu2004_asan_clang_analyzer_patch_3a602b8a22cfedaa2f796fe3580bd0880e2fcee8_657c059fa4cf475058a98567_23_12_18_02_11_31/0/task?bookmarks=0,276) Could you merge the develop branch into your branch when you fix the [comment I mentioned above](https://github.com/wiredtiger/wiredtiger/pull/9932#discussion_r1430311662)? I'll re-review after, but this PR looks good to go after that. Thanks!

hi, @clarissecheah 
I have change the comment and run the newest s_all.
thanks.

**Comment by clarissecheah** [comment]
Hi @y123456yz , it appears as though `s_all` is still failing in the patch build. See below:

```
.947] + echo ####################### MESSAGE ############################
[2023/12/19 09:44:03.947] ####################### MESSAGE ############################
[2023/12/19 09:44:03.947] + echo s_all run of: "./s_clang_format " resulted in:
[2023/12/19 09:44:03.947] s_all run of: "./s_clang_format " resulted in:
[2023/12/19 09:44:03.948] + sed -e s/^/ / __wt.38565
[2023/12/19 09:44:03.948]  Modifying examples/c/ex_all.c
```

 Could I just double check that you've run `s_all` and committed any changes (e.g. in `s_clang_format`)?

**Comment by y123456yz** [comment]
> s_clang_format

hi, @clarissecheah 
I merge the develop and run s_all again,  It should be okay this time.

you can feel free to modify my code branches, If there are any other problems, Please help me with this.

thanks.
:)

**Review by clarissecheah — APPROVED** [review]
LGTM

**Comment by y123456yz** [comment]
the mongo server PR:
https://github.com/mongodb/mongo/pull/1587

**Pull request merged**
